src/hexer/nifcgen.nim

```diff
@@ -498,7 +499,26 @@ proc traverseType(e: var EContext; c: var Cursor; flags: set[TypeFlag] = {}) =
   else:
     error e, "type expected but got: ", c
 
-proc traverseParams(e: var EContext; c: var Cursor) =
+proc maybeByConstRef(e: var EContext; c: var Cursor; pragmas: Cursor) =
+  let param = asLocal(c)
+  if passByConstRef(param.typ, pragmas, sizeof(int)):
```

Comment:
For cross-compiling `sizeof(int)` would be wrong. Hexer needs to understand the `--bits` command line option and Nimony needs to pass it.